trustfall_core/src/interpreter/execution.rs

```diff
@@ -241,6 +241,27 @@ fn construct_outputs<'query, AdapterT: Adapter<'query>>(
     }))
 }
 
+/// Takes a FieldValue and returns a positive usize. If the FieldValue is a Null,
+/// returns None.
```

Comment:
```suggestion
/// Extracts numeric `FieldValue` into a `usize`, clamping negative numbers to 0.
/// Returns `None` on `FieldValue::Null`, and panics otherwise.
```